languages/thingtalk/ast_manip.ts

```diff
@@ -1552,10 +1482,9 @@ function makeComputeFilterExpression(table : Ast.Expression,
 function makeWithinGeoDistanceExpression(table : Ast.Expression, location : Ast.Value, filterValue : Ast.Value) : Ast.Expression|null {
     if (!table.schema!.out.geo || !table.schema!.out.geo.isLocation)
         return null;
-    const filterType = filterValue.getType();
-    if (!(filterType instanceof Type.Measure))
+    if (!(filterValue instanceof Ast.Value.Measure))
         return null;
-    const unit = filterType.unit;
+    const unit = filterValue.unit;
```

Comment:
Why this change? I don't think it's correct if you have a sum of two measurements.

Reply:
I notice you changed this in a previous commit. But the `filterType` unit is always 'm' no matter what's the value's type. We end up generate no within filters.

Reply:
Why do we not generate filters if the unit is m? m is a valid distance unit...

Reply:
when I debug, I find that no matter what the original uni is, the unit from `filterType` is always `m`, and we don't generate numbers over 100

Reply:
Ok so is the problem with min_number/max_number? Those need special handling of "large" numbers, which we generate by proxy generating numbers between 13 and 23

Reply:
No, the issue is that in the following, when we check if the unit in the utterance is at least feet, or the distance is at least 100m, we are not using the actual unit. 
do we really need to have "within a + b miles"?

Reply:
We don't, but I feel like if we don't handle that case correctly it will cause hard to debug issues later on.

Reply:
this code is only for "within" filter; if it's a computation value, it won't break it will just return null, as it should be, because it is not very useful sentence. why will it cause problems in the future...